R/utils.R

```diff
@@ -4,3 +4,23 @@
 file_bug_report_msg <- function() {
   "file a bug report at https://github.com/tntp/surveymonkey/issues"
 }
+
+
+#' Set Survey Monkey Oauth Token
+#'
+#' Sets the option 'sm_oauth_token' for the survey monkey API.
+#' Token will be used for authorization headers for requests.
+#'
+#' @param oauth_token Your survey monkey OAuth 2.0 token.
+#' @export
+set_token = function(oauth_token) {
+  options("sm_oauth_token" = oauth_token)
+}
```

Comment:
I really like this idea and it works great for a _temporary_ token - any way to invoke `usethis::edit_r_profile()` and then prompt the user to paste in `options("sm_oauth_token" = UNIQUE_TOKEN)`? Perhaps with an additional parameter `temporary` = `TRUE/FALSE`?

The way my team uses this package - we have local R installations and only need to set up the .rprofile one time. If we can streamline the setup, it would also be a bonus for the documentation, which users of this package have had issues with before.

Feel free to try to optimize this as much as you'd like, or else I am happy to give it a go when I have the chance. It'd be really cool to walk a new user through the process with in-console comments/directions.

Reply:
You mean on loading the package? if so sure, although not entirely sure how it would work as a temporary thing in the Rprofile at the moment. if you raise an issue for it I can think about it if I get time.

Reply:
although I might add whilst I think something could be done, I don't know if I necessarily agree that it is always a good thing. Package start up messages do clutter up the console.